src/components/tag/Tag.props.ts

```diff
@@ -0,0 +1,15 @@
+import { TTextSizeToken } from '../text/Text.types';
+
+import { TTagVariant } from './Tag.types';
+
+type PTag = {
+    variant: TTagVariant;
+    text: string;
+    size?: TTextSizeToken;
+    onClick?: () => void;
+    className?: string;
+};
+
+export type PTagRoot = Omit<PTag, 'onClick' | 'className'>;
```

Comment:
if you add the `Required<>` utility type you can set default values on the "outer" component and forget about "undefined errors" in the component root

Reply:
that way it is more visible what values are defaults